commit 0211305e7012
Author: Emilio Cobos Álvarez <emilio@crisal.io>
Date:   Fri Sep 22 21:23:11 2017 +0200

    Bug 1402366 - Mark datetimebox as not needing xul.css. r=bholley, r=jessica, a=sledru
    
    We create a datetimebox in content documents that gets an XBL binding on
    nsDateTimeControlFrame::CreateAnonymousContent.
    
    This is done during frame construction, and loading a datetimebox insert xul.css
    in the document, which causes invalidation to run.
    
    datetimebox styles are in html.css, and there's no rule in xul.css that styles
    it, so we don't need to load the stylesheet at all.
    
    MozReview-Commit-ID: 2SbIOoER2uB
    
    --HG--
    extra : source : 5b642e09aacd450f7ea2b667d2d453a701a30e15
---
 dom/xul/nsXULElement.cpp | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git dom/xul/nsXULElement.cpp dom/xul/nsXULElement.cpp
index f1a87e917182..2c3574cd6580 100644
--- dom/xul/nsXULElement.cpp
+++ dom/xul/nsXULElement.cpp
@@ -696,6 +696,7 @@ static inline bool XULElementsRulesInMinimalXULSheet(nsIAtom* aTag)
          aTag == nsGkAtoms::thumb ||
          aTag == nsGkAtoms::scale ||
          // other
+         aTag == nsGkAtoms::datetimebox ||
          aTag == nsGkAtoms::resizer ||
          aTag == nsGkAtoms::label ||
          aTag == nsGkAtoms::videocontrols;
@@ -810,9 +811,7 @@ nsXULElement::BindToTree(nsIDocument* aDocument,
       // pulling in xul.css.
       // Note that add-ons may introduce bindings that cause this assertion to
       // fire.
-      NS_ASSERTION(IsInVideoControls(this) ||
-                   IsInFeedSubscribeLine(this) ||
-                   IsXULElement(nsGkAtoms::datetimebox),
+      NS_ASSERTION(IsInVideoControls(this) || IsInFeedSubscribeLine(this),
                    "Unexpected XUL element in non-XUL doc");
     }
   }

commit 07adf4398955
Author: Emilio Cobos Álvarez <emilio@crisal.io>
Date:   Mon Sep 25 10:16:36 2017 +0200

    Bug 1402366 - Crashtest. r=me, a=sledru
    
    MozReview-Commit-ID: 8ZcVBHKNntj
    
    --HG--
    extra : source : 7fde3082d44b416af788f2ba09834fb3403eb8ff
---
 layout/style/crashtests/1402366.html    | 10 ++++++++++
 layout/style/crashtests/crashtests.list |  1 +
 2 files changed, 11 insertions(+)

diff --git layout/style/crashtests/1402366.html layout/style/crashtests/1402366.html
new file mode 100644
index 000000000000..9cffd08f0415
--- /dev/null
+++ layout/style/crashtests/1402366.html
@@ -0,0 +1,10 @@
+<script>
+document.documentElement.appendChild(document.createElement('x'))
+a = document.createElement('x')
+document.documentElement.appendChild(a)
+new Range().getClientRects()
+document.documentElement.appendChild(document.createElement('tr'))
+b = document.createElement('input')
+b.type = 'date'
+document.documentElement.insertBefore(b, a)
+</script>
diff --git layout/style/crashtests/crashtests.list layout/style/crashtests/crashtests.list
index 9cd557a44187..be05f417d6aa 100644
--- layout/style/crashtests/crashtests.list
+++ layout/style/crashtests/crashtests.list
@@ -203,3 +203,4 @@ load 1387499.html
 load 1391577.html
 load 1398581.html
 load 1402218-1.html
+load 1402366.html
